internal/exec/stages/files/filesystemEntries.go

```diff
@@ -17,15 +17,78 @@ package files
 import (
 	"fmt"
 	"os"
+	"os/exec"
 	"path/filepath"
+	"regexp"
 	"sort"
 	"strings"
 
 	"github.com/coreos/ignition/v2/config/v3_2_experimental/types"
+	"github.com/coreos/ignition/v2/internal/distro"
 	"github.com/coreos/ignition/v2/internal/exec/util"
 	"github.com/coreos/ignition/v2/internal/log"
+
+	"github.com/vincent-petithory/dataurl"
 )
 
+// createCrypttabEntries creates entries inside of /etc/crypttab for LUKS volumes.
+// additionally creates unlock units for clevis based devices.
+func (s *stage) createCrypttabEntries(config types.Config) error {
+	if len(config.Storage.Luks) == 0 {
+		return nil
+	}
+
+	s.Logger.PushPrefix("createCrypttabEntries")
+	defer s.Logger.PopPrefix()
+
+	crypttab := fileEntry{
+		types.Node{
+			Path: "/sysroot/etc/crypttab",
+		},
+		types.FileEmbedded1{},
+	}
+	clevistab := fileEntry{
+		types.Node{
+			Path: "/sysroot/etc/clevistab",
+		},
+		types.FileEmbedded1{},
+	}
```

Comment:
So... I'm thinking we don't use a `clevistab` and custom generator and instead leverage the official Clevis [Password Agent](https://www.freedesktop.org/wiki/Software/systemd/PasswordAgents/): https://github.com/latchset/clevis/blob/0bea5c4f6594a4611498cfbcfe123ed3e092b1e5/src/luks/systemd/clevis-luks-askpass.path. AFAIK, this matches how RHEL/Fedora also use Clevis.

This essentially means that we write everything to `crypttab`.

In practice this works pretty well. I'm working on a patch for upstream to fix some of the dependencies though (this is related to https://github.com/latchset/clevis/issues/54, which is relevant to us, see https://github.com/latchset/clevis/issues/54#issuecomment-549557118).

This is what I'm testing with locally:

```diff
diff --git a/internal/exec/stages/files/filesystemEntries.go b/internal/exec/stages/files/filesystemEntries.go
index 99b573c..a50dbfa 100644
--- a/internal/exec/stages/files/filesystemEntries.go
+++ b/internal/exec/stages/files/filesystemEntries.go
@@ -47,12 +47,6 @@ func (s *stage) createCrypttabEntries(config types.Config) error {
                },
                types.FileEmbedded1{},
        }
-       clevistab := fileEntry{
-               types.Node{
-                       Path: "/sysroot/etc/clevistab",
-               },
-               types.FileEmbedded1{},
-       }
        for _, luks := range config.Storage.Luks {
                dump, err := exec.Command(distro.CryptsetupCmd(), "luksDump", luks.Device).CombinedOutput()
                if err != nil {
@@ -64,28 +58,22 @@ func (s *stage) createCrypttabEntries(config types.Config) error {
                        return fmt.Errorf("couldn't gather luks device %v uuid", luks.Name)
                }
                uuid := string(match[1])
-               if luks.Clevis == nil {
-                       uri := dataurl.EncodeBytes([]byte(fmt.Sprintf("%s UUID=%s %s luks\n", luks.Name, uuid, filepath.Join(util.LuksKeyFilePath, luks.Name))))
-                       crypttab.Append = append(crypttab.Append, types.Resource{
-                               Source: &uri,
-                       })
-               } else {
-                       isNet := "none"
-                       if len(luks.Clevis.Tang) > 0 {
-                               isNet = "_netdev"
-                       }
-                       uri := dataurl.EncodeBytes([]byte(fmt.Sprintf("%s UUID=%s %s\n", luks.Name, uuid, isNet)))
-                       clevistab.Append = append(clevistab.Append, types.Resource{
-                               Source: &uri,
-                       })
+               netdev := ""
+               if len(luks.Clevis.Tang) > 0 {
+                       netdev = ",_netdev"
                }
+               keyfile := "none"
+               if luks.Clevis == nil {
+                       keyfile = filepath.Join(util.LuksKeyFilePath, luks.Name)
+               }
+               uri := dataurl.EncodeBytes([]byte(fmt.Sprintf("%s UUID=%s %s luks%s\n", luks.Name, uuid, keyfile, netdev)))
+               crypttab.Append = append(crypttab.Append, types.Resource{
+                       Source: &uri,
+               })
        }
        if err := crypttab.create(s.Logger, s.Util); err != nil {
                return fmt.Errorf("adding luks devices to crypttab: %v", err)
        }
-       if err := clevistab.create(s.Logger, s.Util); err != nil {
-               return fmt.Errorf("adding luks devices to clevistab: %v", err)
-       }
        return nil
 }

```

Reply:
See Ben's earlier comment https://github.com/coreos/ignition/pull/960#issuecomment-636095257 for why I didn't go down the `clevis-luks-askpass` route.